en/middleware/micrortps.md

```diff
@@ -0,0 +1,344 @@
+  -o AGENTDIR, --agent-outdir AGENTDIR
+                        Agent output dir, by default
+                        src/modules/micrortps_bridge/micrortps_agent
+  -u CLIENTDIR, --client-outdir CLIENTDIR
+                        Client output dir, by default
+                        src/modules/micrortps_bridge/micrortps_client
+  -f FASTRTPSGEN, --fastrtpsgen-dir FASTRTPSGEN
+                        fastrtpsgen installation dir, by default /bin
+  --no-delete           Do not delete dir tree output dir(s)
+```
+
+- The argument `--send/-s` means that the application from PX4 side will send these messages, and the argument `--receive/-r` specifies which messages is going to be received.
+- The output appears in `CLIENTDIR` (`-o src/modules/micrortps_bridge/micrortps_client`, by default) and in the `AGENTDIR` (`-u src/modules/micrortps_bridge/micrortps_agent`, by default). 
+ 
+  > **Caution** This script erases the content of the `CLIENTDIR` and the `AGENTDIR` before creating new files and folders.
+
+- If no flag `-a` or `-c` is specified, both the client and the agent will be generated and installed.
+- The `-f` option may be needed if *Fast RTPS* was not installed in the default location (`-f /path/to/fastrtps/installation/bin`).
+
+An example of use:
+
+```sh
+$ cd /path/to/PX4/Firmware
+$ python Tools/generate_microRTPS_bridge.py -s msg/sensor_baro.msg -r msg/sensor_combined.msg
+```
+
+Checking the correct installation:
+
+```sh
+$ tree src/modules/micrortps_bridge/micrortps_agent
+src/modules/micrortps_bridge/micrortps_agent
+├── build
+├── CMakeLists.txt
+├── idl
+│   ├── sensor_baro_.idl
+│   └── sensor_combined_.idl
+├── microRTPS_agent.cxx
+├── microRTPS_transport.cxx
+├── microRTPS_transport.h
+├── RtpsTopics.cxx
+├── RtpsTopics.h
+├── sensor_baro_.cxx
+├── sensor_baro_.h
+├── sensor_baro_Publisher.cxx
+├── sensor_baro_Publisher.h
+├── sensor_baro_PubSubTypes.cxx
+├── sensor_baro_PubSubTypes.h
+├── sensor_combined_.cxx
+├── sensor_combined_.h
+├── sensor_combined_PubSubTypes.cxx
+├── sensor_combined_PubSubTypes.h
+├── sensor_combined_Subscriber.cxx
+└── sensor_combined_Subscriber.h
+ 2 directories, 20 files
+```
+
+```sh
+$ tree src/modules/micrortps_bridge/micrortps_client
+src/modules/micrortps_bridge/micrortps_client
+├── CMakeLists.txt
+├── microRTPS_client.cpp
+├── microRTPS_transport.cxx
+└── microRTPS_transport.h
+ 0 directories, 4 files
+```
+
+
+## PX4 Firmware: the micro RTPS client
+
+The agent application runs as both a uORB node and as a micro RTPS node. As a uORB node the agent could be subscribed to several topics as well as publish under internal uORB topics. The application receives from a internal publishers the messages, serializes the struct and writes it trough an UART
+device or UDP port selected by the user. Also will be reading from the UART device or UDP port and then publish the info
+to the internal subscribers.
+
+Steps to use the auto generated application:
```

Comment:
The changes to cmake would seem to be some sort of aside or subtopic - no one should ever need to actually do this but we need to keep the information for someone adding the feature at another time to a new target. Do you think we can/should separate this?